biz.aQute.bndlib/src/aQute/bnd/osgi/Processor.java

```diff
@@ -290,7 +290,7 @@ public SetLocation warning(String string, Object... args) {
 		if (!p.warnings.contains(s))
 			p.warnings.add(s);
 		p.signal();
-		return location(s);
+		return p.location(s);
```

Comment:
Great job finding this subtle bug!

Reply:
It's small fries compared with what you guys have done on the rest of the project, so I don't really have a right to accolades.